File: 1-Introduction/Lineup.java
import java.util.*;

class Lineup
{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
		ArrayList<String> v = new ArrayList<>();
		ArrayList<String> incr = new ArrayList<>();
		ArrayList<String> decr = new ArrayList<>();
		for(int i=0; i<n; i++){
			String s = in.next();
			v.add(s);
			incr.add(s);
			decr.add(s);
		}
		Collections.sort(incr);
		Collections.sort(decr, (s1, s2) -> s2.compareTo(s1));
		boolean flg = true;
		for(int i=0; i<n; i++){
			if(!v.get(i).equals(incr.get(i))){
				flg = false;
				break;
			}
		}
		if(flg){
			System.out.println("INCREASING");
			return;
		}
		flg = true;
		for(int i=0; i<n; i++){
			if(!v.get(i).equals(decr.get(i))){
				flg = false;
				break;
			}
		}
		if(flg){
			System.out.println("DECREASING");
			return;
		}
		System.out.println("NEITHER");
	}
}

File: 1-Introduction/lineup.cpp
#include <bits/stdc++.h>
using namespace std;

int n;
int main(){
	cin >> n;
	vector<string> v;
	vector<string> incr;
	vector<string> decr;
	for(int i=0; i<n; i++){
		string s; 
		cin >> s;
		v.push_back(s);
		incr.push_back(s);
		decr.push_back(s);
	}
	sort(incr.begin(), incr.end());
	sort(decr.begin(), decr.end(), greater<string>() );
	
	bool flg = true;
	for(int i=0; i<n; i++){
		if(v[i] != incr[i]){
			flg = false;
			break;
		}
	}
	if(flg){
		cout << "INCREASING\n";
		return 0;
	}
	flg = true;
	for(int i=0; i<n; i++){
		if(v[i] != decr[i]){
			flg = false;
			break;
		}
	}
	if(flg){
		cout << "DECREASING\n";
		return 0;
	}
	cout << "NEITHER";
}
    